client/client_update_test.go

```diff
@@ -893,6 +965,53 @@ func TestUpdateNonRootRemoteCorruptedNoLocalCache(t *testing.T) {
 				role: role,
 			}, testData, true)
 		}
+
+		switch role {
+		case data.CanonicalSnapshotRole:
+			for _, missing := range []string{data.CanonicalRootRole, data.CanonicalTargetsRole} {
+				testData := swizzleExpectations{
+					desc: fmt.Sprintf("snapshot missing root meta checksum"),
+					// TODO: this should probably be always client.ErrMissingMeta
```

Comment:
:(  Validation of the snapshot should really have failed - getting to the syntax error is kind of convoluted.  The error occurs when trying to download and load the root.

- [we get the size and checksum for root from the snapshot](https://github.com/docker/notary/blob/master/tuf/client/client.go#L135)
- there is no file metadata information, so `c.local.Snapshot.Signed.Meta[role]` would have returned the zero value and a `false` for `ok` - since it's a map of structs, it doesn't return `nil` but rather an empty `FileMeta` structure for the zero value
- we don't actually check whether `ok` is true or false, we just grab the `Length`, which is 0 on an empty `FileMeta` object, and the checksum.  An empty `FileMeta` has an empty map for `Hashes`, and getting `"sha256"` from it returns the zero value for that map, which is an empty list -I found this somewhat counter-intuitive, but [that does seem to be the case playing around with empty structs](https://play.golang.org/p/8GZm9WeBc2)
- then we download root using `downloadSigned`, specifying the size to be 0, and end up with an empty list of bytes for root, and because the checksum is empty, we don't bother checking it.
- we take this empty byte array for root and then attempt to JSON-unmarshal it, which results in a syntax error because [you can't unmarshal empty bytes](https://play.golang.org/p/htrncemSUI)